src/eigenlayer/execution.py

```diff
@@ -0,0 +1,196 @@
+    if not tx_receipt['status']:
+        logger.error('queue withdrawal transaction failed')
+        return None
+    return Web3.to_hex(tx)
+
+
+# pylint: disable-next=too-many-arguments
+async def submit_complete_queued_withdrawal_transaction(
+    pod_owner: ChecksumAddress,
+    delegated_to: ChecksumAddress,
+    nonce: int,
+    shares: int,
+    start_block: BlockNumber,
+    receive_as_tokens: bool,
+) -> HexStr | None:
+    """Sends completeQueuedWithdrawal transaction to pod owner contract"""
+    logger.info('Submitting complete queued withdrawal transaction')
+
+    middleware_times_index = (
+        0  # middlewareTimesIndex is unused, but will be used in the Slasher eventually
+    )
+    try:
+        tx = (
+            await EigenPodOwnerContract(pod_owner)
+            .functions.completeQueuedWithdrawal(
+                delegated_to,
+                nonce,
+                shares,
+                start_block,
+                middleware_times_index,
+                receive_as_tokens,
+            )
+            .transact()
+        )
+    except Exception as e:
+        logger.error('Failed to complete queued withdrawal: %s', format_error(e))
+
+        if settings.verbose:
+            logger.exception(e)
+        return None
+
+    logger.info('Waiting for transaction %s confirmation', Web3.to_hex(tx))
+    tx_receipt = await execution_client.eth.wait_for_transaction_receipt(
+        tx, timeout=settings.execution_transaction_timeout
+    )
+    if not tx_receipt['status']:
+        logger.error('complete queued withdrawal transaction failed')
+        return None
+    return Web3.to_hex(tx)
+
+
+async def get_validator_withdrawals_chunk(
+    indexes: set[int], from_block: BlockNumber, to_block: BlockNumber
+) -> list[Withdrawal]:
+    semaphore = asyncio.BoundedSemaphore(EIGEN_VALIDATORS_WITHDRAWALS_CONCURRENCY)
+    pending = {
+        asyncio.create_task(fetch_withdrawals(BlockNumber(block_number), indexes, semaphore))
+        for block_number in range(from_block, to_block + 1)
+    }
+    result = []
+    while pending:
+        done, pending = await asyncio.wait(pending, return_when=asyncio.FIRST_COMPLETED)
+        for task in done:
+            withdrawals = task.result()
+            for withdrawal in withdrawals:
+                result.append(withdrawal)
+
+    return result
+
+
+async def fetch_withdrawals(
+    block_number: BlockNumber, indexes: set[int], semaphore: asyncio.BoundedSemaphore
+) -> list[Withdrawal]:
+    """Fetches block withdrawals."""
+    async with semaphore:
+        block = await execution_client.eth.get_block(block_number)
+        withdrawals = []
+        for index, withdrawal in enumerate(block.get('withdrawals', [])):
+            if int(withdrawal.validatorIndex) in indexes:  # type: ignore[attr-defined]
+                withdrawals.append(
+                    Withdrawal(
+                        block_number=block_number,
+                        validator_index=int(
```

Comment:
`validator_index` should be int as I see in `WithdrawalData` definition. Conversion is not necessary.